engine/src/main/java/org/hibernate/validator/internal/engine/ConfigurationImpl.java

```diff
@@ -132,6 +133,16 @@ private ConfigurationImpl() {
 		this.addConstraintDefinitionContributor( defaultConstraintDefinitionContributor );
 	}
 
+	private ValidatedValueUnwrapper<?> createJavaFXUnwrapperClass(TypeResolutionHelper typeResolutionHelper) {
+		try {
+			Class<?> jfxUnwrapperClass = run( LoadClass.action( JFX_UNWRAPPER_CLASS, getClass().getClassLoader() ) );
+			return (ValidatedValueUnwrapper<?>) ( jfxUnwrapperClass.getConstructor( TypeResolutionHelper.class ).newInstance( typeResolutionHelper ) );
```

Comment:
> org.hibernate.validator itself will never be loaded via the TCCL

`LoadClass#loadClassInValidatorNameSpace()` falls back to the TCCL if a class cannot be loaded through its own loader. Now whether that makes sense...

Tbh. the original fix of only loading the JavaFX unwrapper when its visible to HV's loader seems like the best solution to me. Everything else seems very complex for a tiny potential gain.